pkg/controller/gather_job.go

```diff
@@ -82,17 +83,17 @@ func (d *GatherJob) Gather(ctx context.Context, kubeConfig *rest.Config, protoKu
 
 	// the recorder stores the collected data and we flush at the end.
 	recdriver := diskrecorder.New(d.StoragePath)
-	recorder := recorder.New(recdriver, d.Interval, anonymizer)
-	defer recorder.Flush()
+	rec := recorder.New(recdriver, d.Interval, anonymizer)
 
 	// the gatherers check the state of the cluster and report any
 	// config to the recorder
 	clusterConfigGatherer := clusterconfig.New(gatherKubeConfig, gatherProtoKubeConfig, metricsGatherKubeConfig, anonymizer)
-	err = clusterConfigGatherer.Gather(ctx, configObserver.Config().Gather, recorder)
+	err = clusterConfigGatherer.Gather(ctx, configObserver.Config().Gather, rec)
 	if err != nil {
 		return err
 	}
 
-	klog.Warning("stopped")
+	rec.Flush()
```

Comment:
Why not with `defer`? :)

Reply:
leftover from the previous commit, where I needed a sleep at the very end so I couldn't use defer :)